src/containers/Issue.jsx

```diff
@@ -0,0 +1,58 @@
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import firebase from 'firebase';
+
+import VoteCounter from '../components/VoteCounter'
+
+class Issue extends Component {
+
+  componentDidMount() {
+    const { item, actions } = this.props;
+    const issueRef = firebase.database().ref(`issues/${item.id}`);
+    issueRef.on('value', dataSnapshot => {
+      item.votes = dataSnapshot.val().vote_count ? dataSnapshot.val().vote_count : 0;
+      actions.issueActions.updateIssueVoteCount(item.id, item.votes)
+    });
+  }
+
+  componentWillUnmount() {
+    issueRef.off();
```

Comment:
Because you have one reference to this thing shared by multiple instances of the same class, rather than a reference saved on each instance of the class.

Try something like http://exploringjs.com/es6/ch_classes.html#_static-data-properties